modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -4833,6 +4835,9 @@ private boolean clearLocally0(K key, boolean readers) {
                 deserializeBinary,
                 needVer);
         }
+        catch (IgniteIrreparableConsistencyViolationException e) {
+            throw e;
+        }
         catch (IgniteConsistencyViolationException e) {
             repairAsync(key, ctx.operationContextPerCall(), false).get();
```

Comment:
IMHO, recursion without a limit param is always a bad solution. It looks ok that if `repairAsync` is successful then `repairableGet` will be successfull too. But in one moment due to some bug we can get a StackOverflowException here.

Reply:
IGNITE-16224